matchmake-extension/auto_matchmake_with_param_postpone.go

```diff
@@ -17,9 +17,35 @@ func (commonProtocol *CommonProtocol) autoMatchmakeWithParamPostpone(err error,
 	connection := packet.Sender().(*nex.PRUDPConnection)
 	endpoint := connection.Endpoint().(*nex.PRUDPEndPoint)
 
-	// * A client may disconnect from a session without leaving reliably,
-	// * so let's make sure the client is removed from the session
-	common_globals.RemoveConnectionFromAllSessions(connection)
+	// * Process additionalParticipants early so we can cancel matchmaking if it's wrong
+	// * additionalParticipants is used by Splatoon to move Fest teams around between rooms
+	oldGid := autoMatchmakeParam.GIDForParticipationCheck.Value
+	oldGathering := common_globals.Sessions[oldGid]
+	// * Check the caller is actually in the old session
+	useAdditionalParticipants := oldGathering != nil && oldGathering.ConnectionIDs.Has(connection.ID)
+
+	// * Check the other participants are actually in the old session, and note down the connection IDs
+	additionalParticipants := make([]uint32, autoMatchmakeParam.AdditionalParticipants.Length()+1)
+	// * If using additionalParticipants we'll *move* everyone, rather than disconnect/reconnect.
+	// * This prevents issues from host migration and disconnected notifications.
+	if useAdditionalParticipants {
```

Comment:
some spacing
```suggestion
	// * Check the other participants are actually in the old session, and note down the connection IDs
	additionalParticipants := make([]uint32, autoMatchmakeParam.AdditionalParticipants.Length()+1)

	// * If using additionalParticipants we'll *move* everyone, rather than disconnect/reconnect.
	// * This prevents issues from host migration and disconnected notifications.
	if useAdditionalParticipants {
```